RecoTracker/DeDx/plugins/SiPixelTrackProbQXYProducer.cc

```diff
@@ -0,0 +1,247 @@
+// -*- C++ -*-
+//
+// Package:    SiPixelTrackProbQXYProducer
+// Class:      SiPixelTrackProbQXYProducer
+//
+/**\class SiPixelTrackProbQXYProducer  SiPixelTrackProbQXYProducer.cc RecoTracker/DeDx/plugins/SiPixelTrackProbQXYProducer.cc
+
+ Description: SiPixel Charge and shape probabilities combined for tracks
+
+*/
+//
+// Original Author:  Tamas Almos Vami
+//         Created:  Mon Nov 17 14:09:02 CEST 2021
+//
+
+// system include files
+#include <memory>
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHit.h"
+#include "DataFormats/TrackReco/interface/SiPixelTrackProbQXY.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "TrackingTools/PatternTools/interface/TrajTrackAssociation.h"
+
+//
+// class declaration
+//
+
+class SiPixelTrackProbQXYProducer : public edm::global::EDProducer<> {
+public:
+  explicit SiPixelTrackProbQXYProducer(const edm::ParameterSet&);
+  ~SiPixelTrackProbQXYProducer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+  // ----------member data ---------------------------
+  const bool debugFlag_;
+  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopoToken_;
+  const edm::EDGetTokenT<reco::TrackCollection> trackToken_;
+  const edm::EDPutTokenT<edm::ValueMap<reco::SiPixelTrackProbQXY>> putProbQXYVMToken_;
+  const edm::EDPutTokenT<edm::ValueMap<reco::SiPixelTrackProbQXY>> putProbQXYNoLayer1VMToken_;
+};
+
+using namespace reco;
+using namespace std;
+using namespace edm;
+
+SiPixelTrackProbQXYProducer::SiPixelTrackProbQXYProducer(const edm::ParameterSet& iConfig)
+    : debugFlag_(iConfig.getUntrackedParameter<bool>("debug", false)),
+      tTopoToken_(esConsumes()),
+      trackToken_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks"))),
+      putProbQXYVMToken_(produces<edm::ValueMap<reco::SiPixelTrackProbQXY>>()),
+      putProbQXYNoLayer1VMToken_(produces<edm::ValueMap<reco::SiPixelTrackProbQXY>>("NoLayer1")) {}
+
+void SiPixelTrackProbQXYProducer::produce(edm::StreamID id, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  bool debug = debugFlag_;
```

Comment:
What's the point of copying this variable? Can't the data-member be used directly in the code below?

Reply:
ok, fixed it